llms/mlx_lm/tuner/trainer.py

```diff
@@ -229,9 +250,13 @@ def step(batch):
                 iterate_batches=iterate_batches,
             )
             val_time = time.perf_counter() - stop
-            print(
-                f"Iter {it}: " f"Val loss {val_loss:.3f}, " f"Val took {val_time:.3f}s"
-            )
+            if rank == 0:
+                print(
+                    f"Iter {it}: "
+                    f"Val loss {val_loss:.3f}, "
+                    f"Val took {val_time:.3f}s",
+                    flush=True,
+                )
 
             if training_callback is not None:
```

Comment:
Probably tha tshould go under the `rank==0` condition as well

Reply:
Yeah ok, it makes sense. I was thinking in general callbacks should always run and if the callback is about reporting then it can choose to only run on rank=0. But our callbacks here are only about reporting so it makes sense to just run them only on node 0.

Reply:
It's a good point actually. It's more flexible that way. I'm on board leaving it to the user to specify the rank.